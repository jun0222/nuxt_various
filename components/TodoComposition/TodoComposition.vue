<template>
  <div>
    <h1>Todoリスト</h1>
    <input
      v-model="newTodo"
      placeholder="新しいTodoを追加"
      @keyup.enter="addTodo"
    />
    <ul>
      <li v-for="(todo, index) in todos" :key="index">
        {{ todo }}
        <button @click="removeTodo(index)">削除</button>
      </li>
    </ul>
  </div>
</template>

<script>
import { ref } from '@nuxtjs/composition-api'

export default {
  setup() {
    const todos = ref([])
    const newTodo = ref('')

    const addTodo = () => {
      if (newTodo.value.trim()) {
        todos.value.push(newTodo.value)
        newTodo.value = ''
      }
    }

    const removeTodo = (index) => {
      todos.value.splice(index, 1)
    }

    return {
      todos,
      newTodo,
      addTodo,
      removeTodo,
    }
  },
}
</script>
