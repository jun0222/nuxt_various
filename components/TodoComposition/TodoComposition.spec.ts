import { mount } from '@vue/test-utils'
import TodoComposition from './TodoComposition.vue'
jest.mock('@nuxtjs/composition-api', () => ({
  ref: jest.fn().mockReturnValue({ value: [] }), // モック化したref関数を返す
}))

describe('TodoComposition.vue', () => {
  test('スナップショットテスト', () => {
    const wrapper = mount(TodoComposition)
    expect(wrapper.element).toMatchSnapshot()
  })

  //   test('新しいTodoを追加する正常系テスト', async () => {
  //     const wrapper = mount(TodoComposition)
  //     const input = wrapper.find('input')
  //     const newTodoText = '新しいTodo'

  //     await input.setValue(newTodoText)
  //     await input.trigger('keydown.enter')
  //     await wrapper.vm.$nextTick()
  //     expect(wrapper.text()).toContain(newTodoText)
  //   })
})
