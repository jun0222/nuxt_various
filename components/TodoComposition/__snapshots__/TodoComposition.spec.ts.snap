// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoComposition.vue スナップショットテスト 1`] = `
<div>
  <h1>
    Todoリスト
  </h1>
   
  <input
    placeholder="新しいTodoを追加"
  />
   
  <ul>
    <li>
      
      []
      
      <button>
        削除
      </button>
    </li>
  </ul>
</div>
`;
